Core/include/Acts/Vertexing/ImpactPointEstimator.hpp

```diff
@@ -143,6 +143,39 @@ class ImpactPointEstimator {
       const BoundTrackParameters& track, const Vertex<input_track_t>& vtx,
       const GeometryContext& gctx, const MagneticFieldContext& mctx) const;
 
+  /// @brief Estimates the sign of the 2D and Z lifetime of a given track
+  /// w.r.t. a vertex and a direction (e.g. a jet direction)
+  /// by propagating the trajectory state towards the vertex position
+  /// and computing the scalar product with the direction vector
+  ///
+  /// @param track Track to estimate the IP from
+  /// @param vtx   Vertex the track belongs to
+  /// @param direction   The direction 
+  /// @param gctx  The geometry context
+  /// @param mctx  The magnetic field context
+  ///
+  /// @return A pair holding the sign for the 2D an Z lifetimes
+  Result<std::pair<double,double>>  getLifetimesSignOfTrack(
+      const BoundTrackParameters& track, const Vertex<input_track_t>& vtx,
+      const Acts::Vector3& direction,
+      const GeometryContext& gctx, const MagneticFieldContext& mctx) const;
+
+  /// @brief Estimates the sign of the 3D lifetime of a given track
+  /// w.r.t. a vertex and a direction (e.g. a jet direction)
+  ///
+  /// @param track Track to estimate the IP from
+  /// @param vtx   Vertex the track belongs to
+  /// @param direction   The direction 
+  /// @param gctx  The geometry context
+  /// @param mctx  The magnetic field context
+  ///
+  /// @return A pair holding the sign for the 2D an Z lifetimes
```

Comment:
This only returns a single double.

Reply:
Fixed. However I added a single test unit that checks both 2d and 3d signed lifetimes